Fixes of typesystem issues (PHPStan upgrade preparation)

ondrejmirtes opened the pull request:
cc @PetrHeinz 

| Q             | A
| ------------- | ---
|Description, reason for the PR| Fixes of issues for static analysis by PhpStan (mostly changes in annotations), a follow-up from our discussion on [Shopsys Roadshow 2018 Prague](https://www.shopsys.com/shopsys-roadshow).
|New feature| No <!-- Do not forget to update docs/ -->
|BC breaks| No <!-- Do not forget to update UPGRADE.md -->
|Fixes issues| ...
|Standards and tests pass| No *(note: you can run `php phing standards-fix` in the `php-fpm` container to fix those violations automatically)*
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

CZechBoY commented on project-base/tests/ShopBundle/Database/Model/Product/ProductVisibilityRepositoryTest.php:
```diff
@@ -413,9 +433,13 @@ public function testIsNotVisibleWhenNullManualPrice()
         foreach ($allPricingGroups as $pricingGroup) {
             $productData->manualInputPricesByPricingGroupId[$pricingGroup->getId()] = 10;
+
+
```
why you added 2 blank lines here?

PetrHeinz commented on project-base/tests/ShopBundle/Smoke/NewProductTest.php:
```diff
@@ -66,12 +66,21 @@ private function fillForm(Form $form)
         $form['product_form[descriptionsGroup][descriptions][1]'] = 'test description';
         $form['product_form[pricesGroup][productCalculatedPricesGroup][price]'] = '10000';
-        $form['product_form[pricesGroup][vat]']->select($this->getReference(VatDataFixture::VAT_ZERO)->getId());
+
```
These blank lines visually chunk the form into blocks, but the block do not have any meaning...

Not necessary to change, but imo it would be more readable if you'd first get the necessary objects at the beginning of the method's body, then set the form values.

ondrejmirtes replied:
I'm not sure how I should change it.

vitek-rostislav commented on project-base/tests/ShopBundle/Test/DatabaseTestCase.php:
```diff
@@ -5,7 +5,7 @@
 {
     /**
-     * @return \Doctrine\ORM\EntityManagerInterface
+     * @return \Shopsys\FrameworkBundle\Component\EntityExtension\EntityManagerDecorator
```
We use the interface typehint throughout the application, e.g. in `AdministratorFacade`, even though in `changePassword()` method is  `$this->em->flush()` called an argument. I do agree that it should be changed to the specific implementation but you changed it only in this one place, why?

vitek-rostislav commented on project-base/phpstan.neon:
```diff
@@ -1,5 +0,0 @@
-parameters:
```
the config is used here - https://github.com/shopsys/shopsys/blob/487132e2156d1b701a1eceb5ba079e136172cf5a/project-base/build.xml#L26

and then here - 
https://github.com/shopsys/shopsys/blob/487132e2156d1b701a1eceb5ba079e136172cf5a/project-base/build-dev.xml#L320

but if I get it correctly, it is unnecessary as it does not configure anything. If I am right, please remove the usages mentioned above as well.

PetrHeinz replied:
I would just add that it's used when you download and use `shopsys/project-base` instead of the `shopsys/shopsys` monorepository

ondrejmirtes commented on project-base/app/config/domains_urls.yml:
```diff
@@ -0,0 +1,6 @@
+domains_urls:
```
I added it because other ignored files are also part of the versioned repo, like `parameters.yml`. This one was the only one that was missing.

PetrHeinz replied:
The configuration file `parameters.yml` is not a part of the repo, but it's created during `composer install` automatically (via `Incenteev\ParameterHandler\ScriptHandler::buildParameters` script). We can create some kind of *copy-the-template-if-it-doesn't-exist* script (see #540), but having it included in the repo is not an option.

ondrejmirtes commented:
> The other thing is that I don't fully understand how you made these changes - in particular: Did you do all changes project-wise, or just in some directories? Did you use any automated tools? Did you just fix the problems that did come up when checking with PhpStan? The approach you chose is important for the context of the code review.

I changed `@var` occurences automatically from `/* @var $var Type */` to `/** @var Type $var */` with a regexp. I switched the lines with the variable assignment and the `@var` annotation in all the places manually.

I made sure that all the places PHPStan didn't recognize and complained about are fixed. In practice that means the whole of `project-base`. It's possible that some places still refer to the old way, especially after this rebase, but I don't think it's necessary to fix everything in one go. I'll be continuing sending PRs once this one is merged 😊 I think it's also important to teach your developers what is the correct way of writing `@var` to be recognized by tools like PHPStan. This is it (type can be FQN):

```php
/** @var Type $var */
$var = $this->get('foo');
```

ondrejmirtes commented:
Awesome, thanks 😊 I’ll get back to you with next progress.

On Thu, 25 Oct 2018 at 12:59, Rostislav Vítek <notifications@github.com>
wrote:

> Merged #524 <https://github.com/shopsys/shopsys/pull/524> into master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/shopsys/shopsys/pull/524#event-1925888668>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAGZuM83NUlgfy9wbX1NHUr7OwHa1u3Lks5uoZmcgaJpZM4XZta6>
> .
>
-- 

Ondřej Mirtes
